qendpoint-store/src/main/java/com/the_qa_company/qendpoint/compiler/SparqlRepository.java

```diff
@@ -541,6 +519,37 @@ private ClosableResult<?> execute0(RepositoryConnection customConnection, String
 
 			ParsedQuery parsedQuery = QueryParserUtil.parseQuery(QueryLanguage.SPARQL, sparqlQuery, null);
 
+			// Substitute [AUTO_LANGUAGE] with the client's language
+			if (acceptLanguageHeader != null) {
+				List<Locale.LanguageRange> languageRanges = Locale.LanguageRange.parse(acceptLanguageHeader);
+				Locale locale = Locale.lookup(languageRanges, Arrays.asList(Locale.getAvailableLocales()));
+				try {
+					parsedQuery.getTupleExpr().visit(new AbstractQueryModelVisitor<Exception>() {
+						@Override
+						public void meet(Var node) throws Exception {
+							if (node.getValue() instanceof Literal literal) {
+								boolean isStringDatatype = literal.getDatatype() == null
+										|| literal.getDatatype().equals(CoreDatatype.XSD.STRING.getIri());
+								if (isStringDatatype) {
+									String value = literal.getLabel();
+									if (value.equals("[AUTO_LANGUAGE]")) {
+										ValueFactory vf = SimpleValueFactory.getInstance();
+										Literal newLiteral = vf.createLiteral(locale.getLanguage());
+										Var newVar = new Var(node.getName(), newLiteral);
+										node.replaceWith(newVar);
+									}
+								}
+							}
+							super.meet(node);
+						}
+					});
+				} catch (Exception e) {
+					logger.error("This exception was caught [" + e + "]");
```

Comment:
the logger is here to handle the format (or to avoid it!!!)

the error and warning method accept a throwable (exception) at the end, so you can use

```java
logger.error("This exception was caught", e);
```

In another register you can also use format

```java
logger.error("My number is {}", 42);
```

For primitive values it won't make a big difference, but for large object (like exception), calling or not the toString will make one.